trailblazer/services/slurm/slurm_api_service/slurm_api_service.py

```diff
@@ -0,0 +1,24 @@
+from trailblazer.clients.slurm_api_client.dto.job_response import SlurmJobResponse
+from trailblazer.clients.slurm_api_client.slurm_api_client import SlurmAPIClient
+from trailblazer.services.slurm.dtos import SlurmJobInfo
+from trailblazer.services.slurm.slurm_api_service.utils import create_job_info_dto
+from trailblazer.services.slurm.slurm_service import SlurmService
+
+
+class SlurmAPIService(SlurmService):
+    def __init__(self, client: SlurmAPIClient):
+        self.client = client
+
+    def get_job(self, job_id: int) -> SlurmJobInfo | None:
+        job_response: SlurmJobResponse = self.client.get_job(job_id)
+        if not job_response.jobs or job_response.errors:
+            return None
+        return create_job_info_dto(job_response)
+
+    def get_jobs(self, job_ids: list[int]) -> list[SlurmJobInfo]:
+        jobs: list[SlurmJobInfo] = []
+        for job_id in job_ids:
+            job: SlurmJobInfo | None = self.get_job(job_id)
+            if job:
+                jobs.append(job)
```

Comment:
If we want to make this efficient I think we might want to change the logic. I think it might be faster to iterate over all list and check if they exist in job_ids rather than iterating through the small one and referencing the big one.

Feel free to leave this since we're not using this function atm

Reply:
I don't understand this suggestion! Do you mean that we should retrieve information about all jobs currently pending/running by SLURM and use that list?

That could potentially be more efficient since it would decrease the number of requests. I don't think this part of the code is performance critical though, so optimizing it now seems unnecessary - especially since it is not used like you say.

Reply:
Yeah, I meant fetching all jobs and then iterating through them and picking out the jobs in job_ids.

Let's skip it for now

Reply:
I think that implementation makes more sense, but lets leave it for now